### DISCUSSION TITLE
v20.backup.realm after migration

### DISCUSSION THREAD

**Discussion by vardansargsyan92** [realm/realm-dotnet#2523] [Q&A] [answered]
Hello.

When migrating the old database, for example, `test_database.realm` new realm file is created  `test_database.v20.backup.realm` by SDK. I guess the SDK backups the old database before starting the migration. So the question is does the SDK remove the backup file or should be doing it manually from our side?

**Comment by DominicFrei** [comment, marked as answer]
Hi @vardansargsyan92!

Backup files will be created whenever a file format change happened. In your case a backup of version 20 was created. They will be kept for 3 months to make sure you do have a backup in case of any problems. After 3 months they will be automatically deleted.

Migrations do not create any backups. I assume you did a migration and also had a file format change. But they are not connected to each other.

**Reply by vardansargsyan92** [reply]
hi @DominicFrei.

In my case, we have an app backup mechanism with the current database scheme version. Suppose we have the application with a 1.0 database schema version. And suppose that we have backed up our app with the current database.  We do some database refactoring, implement corresponding migration functions, and increment the schema version to 2.0. Now, when I try to restore the backup with the 1.0 version according to the latest changes the migration process starts, and when it finishes a v20.backup.realm file is created. I was wondering that it would be somehow be deleted by SDK, but it doesn't unless a 3 month is over. 

So in my case, that file causes an exception when I clear the current folder. Thus I need to delete it immediately after successfully restoring backups.

**Reply by DominicFrei** [reply]
Ah, I understand!

Well, you can always delete it manually.

Just out of curiosity: how do you delete the current folder? Do you use `RemoveRealm` or is it done manually?

**Reply by vardansargsyan92** [reply]
we do it manually

**Reply by DominicFrei** [reply]
Ok. What kind of exception do you run into when deleting the folder manually? I assume it's just a recursive delete of whatever is in the folder? Trying to figure out if there is also a problem on our side that we need to address.

**Reply by vardansargsyan92** [reply]
actually the exception is thrown on these lines of code (It's not related to the delete process)

`string dbPath = (await _fileService.GetFilesAsync(dbFolderPath, "*.realm")).SingleOrDefault();`
`var database = Database.GetDatabaseFromPath(dbPath, _fileService, _dateTimeProvider);`
We assume that the current folder contains a single realm file and that assumption was correct until the Realm's package update. Now we get two realm files and need to delete  the v20.backup.realm at some point of database fetching cycle.

**Reply by DominicFrei** [reply]
Ok, understood. Glad to see you know where the problem is and how to fix it. 👍

**Reply by vardansargsyan92** [reply]
Thanks a lot. The support's instant responses are accelerating my working productivity.

**Reply by DominicFrei** [reply]
🙃
